test/blackbox-tests/test-cases/merlin/implicit-transitive-deps.t/run.t

```diff
@@ -3,31 +3,36 @@
 
   $ dune build @check
 
-FIXME: Merlin must be able to locate the definitions of values coming from an
+Merlin must be able to locate the definitions of values coming from an
 implicit transitive dependency, even when `implicit-transitive-dependency` is
 set to `false`. They should be part of the source path returned by the
 configuration.
 
 In this test the dependencies are as follow:
 main -> lib1 -> lib2 -> stdlib
 
+When using OCaml < 5.2 there is no proper way to provide that information to
+Merlin.
   $ FILE=$PWD/bin/main.ml
   $ printf "(4:File%d:%s)" ${#FILE} $FILE | dune ocaml-merlin |
-  > sed -E "s/[[:digit:]]+:/\?:/g" | tr '(' '\n' | grep ":S?"
+  > sed -E "s/[[:digit:]]+:/\?:/g" | tr '(' '\n' | grep -E ":[BS]H?\?"
+  ?:B?:$TESTCASE_ROOT/_build/default/bin/.main.eobjs/byte)
+  ?:B?:$TESTCASE_ROOT/_build/default/src/lib1/.lib1.objs/byte)
   ?:S?:$TESTCASE_ROOT/bin)
   ?:S?:$TESTCASE_ROOT/src/lib1)
-  ?:S?:$TESTCASE_ROOT/src/lib2)
-  ?:S?:$TESTCASE_ROOT/src/lib_dep)
```

Comment:
@Leonidas-from-XIV  I will have a look on Monday, but I think this is a regression for users of OCaml <5.2

Reply:
Actually, since `lib2` is hidden, it should not be a part of the standard source path for Merlin, this was a regression introduced with the indexing rules, and the current behavior should be the correct one. The only correct way to handle these are via the new `SH` and `BH` directives, which are specific to 5.2.